railties/lib/rails/commands/dbconsole/dbconsole_command.rb

```diff
@@ -52,6 +52,11 @@ def start
         ENV["PGSSLCERT"]      = config[:sslcert].to_s if config[:sslcert]
         ENV["PGSSLKEY"]       = config[:sslkey].to_s if config[:sslkey]
         ENV["PGSSLROOTCERT"]  = config[:sslrootcert].to_s if config[:sslrootcert]
+        if config[:variables]
+          ENV["PGOPTIONS"] = config[:variables].filter_map do |name, value|
+            "-c #{name}=#{value.gsub(/[ \\]/, '\\\\\0')}" unless value == ":default" || value == :default
```

Comment:
Sorry I didn't think about this before, but shouldn't we call `to_s` on `value`?

```suggestion
            "-c #{name}=#{value.to_s.gsub(/[ \\]/, '\\\\\0')}" unless value == ":default" || value == :default
```

I assume there may be some config options which, even if they're meant to be specified as a string, could be written in YAML and parsed as e.g. a number or boolean.

Reply:
Yes, sure, this is an oversight from me too. Also updated a test case.